### ISSUE TITLE
Can't make it train

### ISSUE THREAD

**Issue by sw1sh** [flipkart-incubator/fk-visual-search#1]
Can you write a more detailed instructions on how to run the training with street2shop, please? I've managed to prepare the dataset, fix all the errors, and actually start the solver. But the loss is not going down, it's stuck on 0.018. Am I forgetting something? I'm not familiar with caffe very much, but it seems pretty straightforward. Some images were missing, so that also required some clearing up to do. 

```
I0405 21:17:02.524794 10864 solver.cpp:219] Iteration 9000 (0.480324 iter/s, 2081.93s/1000 iters), loss = 0.018
I0405 21:17:02.525902 10864 solver.cpp:238]     Train net output #0: diff_disp = 0
...
I0405 21:17:02.526015 10864 solver.cpp:238]     Train net output #19: diff_disp = 0
I0405 21:17:02.526036 10864 solver.cpp:238]     Train net output #20: diff_gap_disp = 0.0009
...
I0405 21:17:02.526154 10864 solver.cpp:238]     Train net output #39: diff_gap_disp = 0.0009
I0405 21:17:02.526187 10864 solver.cpp:238]     Train net output #40: hinge_loss = 0.0009 (* 1 = 0.0009 loss)
...
I0405 21:17:02.526360 10864 solver.cpp:238]     Train net output #59: hinge_loss = 0.0009 (* 1 = 0.0009 loss)
I0405 21:17:02.526383 10864 solver.cpp:238]     Train net output #60: label_n = 0
...
I0405 21:17:02.526481 10864 solver.cpp:238]     Train net output #79: label_n = 0
I0405 21:17:02.526504 10864 solver.cpp:238]     Train net output #80: label_p = 0
...
I0405 21:17:02.526610 10864 solver.cpp:238]     Train net output #99: label_p = 0
I0405 21:17:02.526630 10864 solver.cpp:238]     Train net output #100: label_q = 0
...
I0405 21:17:02.526728 10864 solver.cpp:238]     Train net output #119: label_q = 0
I0405 21:17:02.526748 10864 solver.cpp:238]     Train net output #120: neg_dist_disp = 0
I...
I0405 21:17:02.526854 10864 solver.cpp:238]     Train net output #139: neg_dist_disp = 0
I0405 21:17:02.526873 10864 solver.cpp:238]     Train net output #140: pos_dist_disp = 0
...
I0405 21:17:02.526971 10864 solver.cpp:238]     Train net output #159: pos_dist_disp = 0
```

I've changed the [model](https://gist.github.com/anonymous/46df509604d8b3d574e23f276c3c895a) a bit, put a Reshape layer (caffe threw some errors about not be able to reshape inplace) and image resizing before crop. The solver is essentially the same.

**Comment by ananya-h-a** [comment]
The network is converging to gap. This usually happens with bad initialisation. Did you finetune from VGG- 16 when you trained?

**Comment by sw1sh** [comment]
I think I did, but maybe not the right way. This was my way to finetune:

```
vgg16 = caffe.Net('vgg/deploy.prototxt', 'vgg/VGG_ILSVRC_16_layers.caffemodel', caffe.TEST)

for triplet in ['q', 'p', 'n']:
    for layer, blob in vgg16.blobs.iteritems():
        if re.match('(conv|pool|fc).+', layer):
            feature_extractor.net.blobs[layer + '_' + triplet] = blob
```

**Comment by ananya-h-a** [comment]
Are you training from Python? 
If you're using the command line, use  -weights argument with the path to the VGG-16 model.

**Comment by sw1sh** [comment]
Yeah, I've saved the net after reassigning those blobs and used it in -weights.

**Comment by ananya-h-a** [comment]
You don't need to do that. You can directly run using -weights. 
Can you check your logs to see if the weights are being initialized properly? You should be able to see what weights are being used for every layer in the logs.

**Comment by sw1sh** [comment]
Yeah! It worked, but complained about fc6 layer due to shape mismatch, so I've renamed it, and it looks like the loss is coming down now!

**Comment by sw1sh** [comment]
Well, actually it didn't. The loss was changing a little, but still essentially at 0.018 and embedding layer produces the same vector for every image I've tried. Can you share your pretrained model by any chance?

**Comment by vj-1988** [comment]
@sw1sh, @ananya-h-a  : Can you explain me more on how to prepare train_q and test_q to train? I followed all the steps mentioned and I ended up with a csv file containing filepath of the triplet files.

**Comment by Usernamezhx** [comment]
@sw1sh hi sw1sh  sorry to bother you .why do you change the input image size?
